app/src/main/java/com/example/hris/ui/viewmodels/LoginViewModel.kt

```diff
@@ -19,16 +23,24 @@ class LoginViewModel @Inject constructor(
     val loadingDialogState = MutableLiveData<Boolean>()
     val message = MutableLiveData<String>()
 
-    fun userLogin(username: String, password: String) {
+    fun userLogin(username: String, password: String, context: Context) {
```

Comment:
Avoid passing the Context object to ViewModels as the ViewModels have a longer lifecycle than Fragments/Activities which hosts the Context object.

If you are just accessing the Resources, just use the Application object from the AndroidViewModel constructor